### ISSUE TITLE
AUC score on training data is not computed

### ISSUE THREAD

**Issue by AlexGrig** [catboost/catboost#450]
Hi,
I have small example where the AUC score is not computed at all on the trainig data:

```
import catboost
from catboost.datasets import titanic
import numpy as np
from sklearn.model_selection import train_test_split

# data preparation
train_df, test_df = titanic()
train_df.fillna(-999, inplace=True)
test_df.fillna(-999, inplace=True)
X, y = train_df.drop('Survived', axis=1), train_df.Survived
categorical_features_indices = np.where(X.dtypes != np.float)[0]
X_train, X_validation, y_train, y_validation = train_test_split(X, y, 
                                      test_size=0.3, random_state=40, shuffle = True)

# training
params = { 'iterations': 7, 'loss_function': 'CrossEntropy', 
           'eval_metric': 'AUC', 'custom_metric': ['AUC','Precision', 'Recall'], 
          'logging_level': 'Verbose', 
          'use_best_model': False, 'random_seed': 0,
          'train_dir': './titanic_experiment'}

train_pool = catboost.Pool(X_train, y_train, 
                           cat_features=categorical_features_indices)
validate_pool = catboost.Pool(X_validation, y_validation, 
                          cat_features=categorical_features_indices)
model = catboost.CatBoostClassifier(**params)
model.fit(train_pool, eval_set=validate_pool, verbose=True)
```

The output of this is:

```
Learning rate set to 0.5
0:	test: 0.8466117	best: 0.8466117 (0)	total: 21ms	remaining: 126ms
1:	test: 0.8631811	best: 0.8631811 (1)	total: 29.1ms	remaining: 72.8ms
2:	test: 0.8631811	best: 0.8631811 (1)	total: 39.1ms	remaining: 52.1ms
3:	test: 0.8798363	best: 0.8798363 (3)	total: 49ms	remaining: 36.8ms
4:	test: 0.8830701	best: 0.8830701 (4)	total: 57.6ms	remaining: 23.1ms
5:	test: 0.8840144	best: 0.8840144 (5)	total: 70.1ms	remaining: 11.7ms
6:	test: 0.8958047	best: 0.8958047 (6)	total: 81.5ms	remaining: 0us

bestTest = 0.8958047161
bestIteration = 6
```

As you can see the train AUC metric is not shown in the output. I also checked the folder, it is also absent. If I plug F1 as an eval_metric is works properly. What could be the reason?
 
catboost version: '0.9.1.1'
Operating System: 'MacOS 10.13.4'
CPU: '2 GHz Intel Core i7'

**Comment by annaveronika** [comment]
This is expected behavoir:
https://tech.yandex.com/catboost/doc/dg/concepts/loss-functions-docpage/

We don't compute AUC, PFound and NDCG by default on training data, because computation of these metrics is very slow and is the bottleneck of the training.
If you still want to compute them, you need to use AUC:hints=skip_train~false as described in documentation.

I suggest also to use metric_period=50 in this case, because calculating AUC on every iteration is too expensive.

**Comment by AlexGrig** [comment]
Ok, thank you!

**Comment by fingoldo** [comment]
Wondering, is there an easy way to set skip_train~true somehow for a custom eval_metric (specified by a class)?
